networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -175,11 +184,11 @@ public void onIncomingMessage(
                   new RequestState(
                       callback,
                       startSlot,
-                      endSlot,
+                      endSlotBeforeFulu,
                       canonicalHotRoots,
                       finalizedSlot,
                       specConfig.getMaxRequestBlobSidecars());
-              if (message.getCount().isZero()) {
+              if (message.getCount().isZero() || !availabilityRequiredAtStart) {
```

Comment:
not sure about this.
Say, (forget about fulu) we are at epoch 5096, requested slot at epoch 999 with count 100, so we should reply with some blobSidecars starting from epoch 1000, but we will answer with empty response in this case. Also I'd cover this case with a test